### ISSUE TITLE
Python interface errors and needs more tests.

### ISSUE THREAD

**Issue by henrykironde** [weecology/retriever#1263]
We have currently added the `data_dir` parameter. This has caused some changes that introduced some errors. 

```python
Import retriever as rt
rt.install_json("iris", data_dir="test)
Installing iris_Iris.json: 100%|██████████████████████████████████████████| 151/151 [00:00<00:00, 18683.70rows/s]
[Errno 2] No such file or directory: 'iris_Iris.json'
```

To solve this we need to rewrite the disconnect functions for some of the engines like xml, json,

**Comment by ethanwhite** [comment]
This seems to be working for me on `master`:

```python
In [1]: import retriever as rt                                                                                                                                                                                     

In [2]: rt.install_json('iris')                                                                                                                                                                                    
Downloading bezdekIris.data: 3.00B [00:00, 14.3B/s]                                                                                                                                                                
Installing iris_Iris.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 151/151 [00:00<00:00, 14006.06rows/s]
Out[2]: <retriever.engines.jsonengine.engine at 0x7f038cf8c978>

In [3]: rt.install_json('iris', data_dir='mytest')                                                                                                                                                                 
Installing iris_Iris.json:  50%|██████████████████████████████████████████████████████████████████████▏                                                                     | 151/301 [00:00<00:00, 22524.36rows/s]
Out[3]: <retriever.engines.jsonengine.engine at 0x7f038cf8c978>
```

**Comment by henrykironde** [comment]
Thanks Ethan, I will do a clean test and let you know

**Comment by henrykironde** [comment]
Am still getting the error 

```python

Python 3.6.5 (default, Apr 25 2018, 14:26:36) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import retriever as rt 
>>> rt.install_json('iris', data_dir='mytest') 
Installing iris_Iris.json: 100%|██████████████████████████████████████████| 151/151 [00:00<00:00, 19701.98rows/s]
[Errno 2] No such file or directory: 'iris_Iris.json'
<retriever.engines.jsonengine.engine object at 0x115ca5e80>
>>> 
```

**Comment by Jaffrey98** [comment]
@henrykironde I'm not getting the error too, just like @ethanwhite
The package I'm using was **installed from source (master)**
Platform: Windows 10 (64 bit)

```python
D:\Projects\GSOC>python
Python 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 17:00:18) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import retriever as rt
>>> rt.install_json('iris')
Installing iris_Iris.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 151/151 [00:00<00:00, 22146.30rows/s]
<retriever.engines.jsonengine.engine object at 0x0000011F957305C0>
>>> rt.install_json('iris',data_dir="test")
Installing iris_Iris.json:  50%|██████████████████████████████████████████████████████▋                                                      | 151/301 [00:00<00:00, 25778.01rows/s]
<retriever.engines.jsonengine.engine object at 0x0000011F957305C0>
>>>
```

Although I don't know why its showing 50% when i added the "data_dir" argument even though the file was fully downloaded inside the directory.

**Comment by henrykironde** [comment]
Thanks for the test @Jaffrey98, That is probably from running `tqdm` with out having asleep. But it does not affect the functionality of the retriever.
We could open an issue, to find a better way to represent the bars well.

**Comment by ethanwhite** [comment]
OK, I did see a version of this error a minute ago. I think it's related to #1264 as @henrykironde noted. Don't have a fully reproducible example yet, but wanted to say that I'd seen it happen.

**Comment by ethanwhite** [comment]
OK, I've got this one on the run. The new `data_dir` didn't get added to the `table_names` list that we're using for `disconnect`. So, not related to #1264 and @henrykironde had the source of the error right (no surprise). The reason @Jaffrey98 and I didn't see the error is because we ran the install without `data_dir` first, which put the file in the working directory, which allow `disconnect` to work.

@henrykironde - I'm working on this now. I'm guessing I should expect the same issue for the other flat file engines?
